activesupport/test/message_encryptor_test.rb

```diff
@@ -239,6 +241,82 @@ def munge(base64_string)
     end
 end
 
+class DefaultMarshalSerializerMessageEncryptorTest < MessageEncryptorTest
+  def setup
+    @secret    = SecureRandom.random_bytes(32)
+    @verifier  = ActiveSupport::MessageVerifier.new(@secret, serializer: ActiveSupport::MessageEncryptor::NullSerializer)
+    @encryptor = ActiveSupport::MessageEncryptor.new(@secret)
+    @data = { "some" => "data", "now" => Time.local(2010) }
+    @default_fallback_to_marshal_serialization = ActiveSupport::JsonWithMarshalFallback.fallback_to_marshal_serialization
+    ActiveSupport::JsonWithMarshalFallback.fallback_to_marshal_serialization = true
+  end
+
+  def teardown
+    ActiveSupport::JsonWithMarshalFallback.fallback_to_marshal_serialization = @default_fallback_to_marshal_serialization
+    super
+  end
+
+  def test_backwards_compat_for_64_bytes_key
+    # 64 bit key
+    secret = ["3942b1bf81e622559ed509e3ff274a780784fe9e75b065866bd270438c74da822219de3156473cc27df1fd590e4baf68c95eeb537b6e4d4c5a10f41635b5597e"].pack("H*")
+    # Encryptor with 32 bit key, 64 bit secret for verifier
+    encryptor = ActiveSupport::MessageEncryptor.new(secret[0..31], secret)
+    # Message generated with 64 bit key
+    message = "eHdGeExnZEwvMSt3U3dKaFl1WFo0TjVvYzA0eGpjbm5WSkt5MXlsNzhpZ0ZnbWhBWFlQZTRwaXE1bVJCS2oxMDZhYVp2dVN3V0lNZUlWQ3c2eVhQbnhnVjFmeVVubmhRKzF3WnZyWHVNMDg9LS1HSisyakJVSFlPb05ISzRMaXRzcFdBPT0=--831a1d54a3cda8a0658dc668a03dedcbce13b5ca"
+    assert_equal "data", encryptor.decrypt_and_verify(message)[:some]
+  end
+
+  def test_on_rotation_is_called_and_returns_modified_messages
+    older_message = ActiveSupport::MessageEncryptor.new(secrets[:older], "older sign").encrypt_and_sign({ encoded: "message" })
+
+    encryptor = ActiveSupport::MessageEncryptor.new(@secret)
+    encryptor.rotate secrets[:old]
+    encryptor.rotate secrets[:older], "older sign"
+
+    rotated = false
+    message = encryptor.decrypt_and_verify(older_message, on_rotation: proc { rotated = true })
+
+    assert_equal({ "encoded" => "message" }, message)
```

Comment:
Expecting this to be `assert_equal({ encoded: "message" }, message)`, since this test class' goal is to make it so `Marshal` serialization is used.